### PR TITLE
[1/2] Add golang apache livy client

### PR DESCRIPTION
**Pull request by hsubbaraj-spiral**
## Describe your changes and why you are making these changes
Add Livy client that will be used by SparkJobManager to ship operators to Spark EMR.
## Checklist before requesting a review
- [x] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [x] I have performed a self-review of my code.
- [ ] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [ ] If this is a new feature, I have added unit tests and integration tests.
- [ ] I have run the integration tests locally and they are passing.
- [ ] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [ ] All features on the UI continue to work correctly.
- [ ] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

File: src/golang/lib/livy/client_test.go [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+	mux.HandleFunc("/sessions", func(w http.ResponseWriter, r *http.Request) {
+		jsonResp := `{"id": 1, "state": "idle"}`
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusCreated)
+		w.Write([]byte(jsonResp))
+	})
+
+	// Call CreateSession with expected
+	sessionReq := &CreateSessionRequest{
+		Kind: "pyspark",
```
Comment (likawind) [inline src/golang/lib/livy/client_test.go:42]: does these test require an active spark cluster?

File: src/golang/lib/livy/entrypoints.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+const (
+	InstallAqueductEntrypoint = `import sys
+import subprocess
+
+print(1)
+subprocess.check_call([sys.executable, '-m', 'pip', 'install', 'aqueduct-ml==0.2.1'])
+`
+
+	FunctionEntrypoint = `import base64
```
Comment (likawind) [inline src/golang/lib/livy/entrypoints.go:11]: How are these entry points being used?

File: src/golang/lib/livy/types.go [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+	Available      StatementState = "available"
+	StatementError StatementState = "error"
+	Cancelling     StatementState = "cancelling"
+
+	OK    StatementOutputStatus = "ok"
+	Error StatementOutputStatus = "error"
+)
+
+// Livy Session.
+type Session struct {
```
Comment (likawind) [inline src/golang/lib/livy/types.go:31]: Where are these statements come from? Looks like there's some public reference on how to define those?

**Review by likawind — APPROVED** [review]
Let's get the functional part working as long as it didn't break main. I think we can merge this change, but I'm curious what's the difference between our own library vs the go API: https://pkg.go.dev/github.com/bilcus/livy-api-go-client .

File: src/golang/lib/livy/entrypoints.go (hunk 2 above)
Comment (hsubbaraj-spiral) [reply src/golang/lib/livy/entrypoints.go:11]: These entrypoints are the base code that are being shipped to Spark. We inject the spec into the entrypoint before shipping.

File: src/golang/lib/livy/types.go (hunk 3 above)
Comment (hsubbaraj-spiral) [reply src/golang/lib/livy/types.go:31]: https://livy.incubator.apache.org/docs/latest/rest-api.html

Comment (likawind) [reply src/golang/lib/livy/types.go:31]: I see, let's ship this version and get the feature done first. I do feel we can separately explore the livy go package mentioned above, as it appear to have everything implemented

File: src/golang/lib/livy/client_test.go (hunk 1 above)
Comment (hsubbaraj-spiral) [reply src/golang/lib/livy/client_test.go:42]: No, these use the `httptest` package to mock the server.

File: src/golang/lib/livy/entrypoints.go [hunk 4]
```diff
@@ -0,0 +1,65 @@
+package livy
```
Comment (saurav-c) [inline src/golang/lib/livy/entrypoints.go:1]: Do you see a future refactor where we might put this python code in our python codebase so it's easier to maintain?

Comment (hsubbaraj-spiral) [reply src/golang/lib/livy/entrypoints.go:1]: Yep, I have to investigate the best way to do this. We want to ship this code as a string and need it available from the server binary. Not sure if the server binary has access to aqueduct-ml package.

**Pull request merged**